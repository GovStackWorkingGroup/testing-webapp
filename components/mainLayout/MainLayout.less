@import '../../styles/variables.less';

.main-layout {
  background-color: @background;
  width: 100%;
  height: 100%;
}
