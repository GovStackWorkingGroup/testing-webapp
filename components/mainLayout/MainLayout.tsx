import React from 'react';

type MainLayoutProps = {
  children: React.ReactNode
}

const MainLayout = ({ children }: MainLayoutProps) => (
  <div className="main-layout">{children}</div>
);

export default MainLayout;
