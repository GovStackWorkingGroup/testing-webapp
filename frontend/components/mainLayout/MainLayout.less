@import '../../styles/Variables.less';

.main-layout {
  width: 100%;
  height: calc(100vh + 10px);
}
