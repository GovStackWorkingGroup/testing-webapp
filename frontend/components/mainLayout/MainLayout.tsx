import React from 'react';
import Header from '../header/Header';

type MainLayoutProps = {
  children: React.ReactNode
}

const MainLayout = ({ children }: MainLayoutProps) => (
  <div className="main-layout">
    <Header/>
    {children}
  </div>
);

export default MainLayout;
