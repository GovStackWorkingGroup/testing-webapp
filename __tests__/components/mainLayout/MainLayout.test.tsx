import MainLayout from '../../../components/mainLayout/MainLayout';
import { render } from '../../test-utils/test-utils';

describe('Unit test for MainLayout component', () => {
  const childrenExample = <p>Test</p>;

  it('render the MainLayout component', () => {
    const { container } = render(<MainLayout>{childrenExample}</MainLayout>);

    expect(container).toMatchSnapshot();
  });
});
