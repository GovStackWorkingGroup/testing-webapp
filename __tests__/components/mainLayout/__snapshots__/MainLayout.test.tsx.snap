// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for MainLayout component render the MainLayout component 1`] = `
<div>
  <div
    class="main-layout"
  >
    <div
      class="header"
    >
      <div
        class="header-logo"
        data-testid="logo"
      >
        <img
          alt="logo"
          src="/images/logo.png"
        />
      </div>
    </div>
    <p>
      Test
    </p>
  </div>
</div>
`;
